cfme/tests/services/test_orchestration_template.py

```diff
@@ -47,82 +49,138 @@
 @pytest.fixture(scope="function")
-def create_template():
-    method = METHOD_TORSO.replace('CloudFormation', fauxfactory.gen_alphanumeric())
-    return method
+def template_type(request):
+    return request.param
 
 
-def test_orchestration_template_crud(appliance, provisioning, create_template):
-    template_type = provisioning['stack_provisioning']['template_type']
-    temp_type = provisioning['stack_provisioning']['template_type_dd']
+@pytest.yield_fixture(scope="function")
+def created_template(appliance, template_type):
+    method = METHOD_TORSO.replace('CloudFormation', fauxfactory.gen_alphanumeric())
     collection = appliance.collections.orchestration_templates
-    template = collection.create(template_type=template_type, temp_type=temp_type,
+    template = collection.create(template_group=templates.get(template_type)[1],
+                                 template_type=templates.get(template_type)[0],
                                  template_name=fauxfactory.gen_alphanumeric(),
-                                 description="my template", content=create_template)
+                                 description="my template",
+                                 content=method)
+    yield template
+    template.delete()
+
 
+def test_orchestration_template_crud(appliance, template_type):
+    method = METHOD_TORSO.replace('CloudFormation', fauxfactory.gen_alphanumeric())
+    collection = appliance.collections.orchestration_templates
+    template = collection.create(template_group=templates.get(template_type)[1],
+                                 template_type=templates.get(template_type)[0],
+                                 template_name=fauxfactory.gen_alphanumeric(),
+                                 description="my template",
+                                 content=method)
+    assert template.exists
     with update(template):
         template.description = "my edited description"
     template.delete()
+    assert not template.exists
 
 
-def test_copy_template(appliance, provisioning, create_template):
-    template_type = provisioning['stack_provisioning']['template_type']
-    temp_type = provisioning['stack_provisioning']['template_type_dd']
-    collection = appliance.collections.orchestration_templates
-    template = collection.create(template_type=template_type, temp_type=temp_type,
-                                 template_name=fauxfactory.gen_alphanumeric(),
-                                 description="my template", content=create_template)
+def test_copy_template(created_template):
+    """Tests Orchestration template copy"""
     copied_method = METHOD_TORSO_copied.replace('CloudFormation', fauxfactory.gen_alphanumeric())
-    template.copy_template(template.template_name + "_copied", copied_method)
-    template.delete()
+    template = created_template
+    template_copy = template.copy_template(template.template_name + "_copied", copied_method)
+    assert template_copy.exists
+    template_copy.delete()
```

Comment:
**QUESTION:** Whenever I copy something the main quality I want to know about the copy is its equivalency to the thing from which it is copied.   Is there anyway you could certain things are the same between the copies?   Or would that really be testing the provider not CFME's use of the provider?

Reply:
Copied template won't be equal. another test verifies that - content must be unique. when we copy orchestration template - we just use original template as draft/base template - but we have to modify name/content. only type remains. additionally there's there's no relationship between templates to get parent template.